tests/conftest.py

```diff
@@ -0,0 +1,41 @@
+import os
+import pytest
+from unittest.mock import patch
+from sys import modules
+
+def remove_module_cache():
+    for module in (
+        'pdir',
+        'pdir.api',
+        'pdir.constants',
+        'pdir.format',
+        'pdir.configuration',
+        'pdir.color',
+        'pdir.utils',
+    ):
+        try:
+            del modules[module]
+        except KeyError:
+            pass
+
+
+@pytest.fixture
+def clean():
+    remove_module_cache()
+
+    DEFAULT_CONFIG_FILE = os.path.join(os.path.expanduser('~'), '.pdir2config')
+    yield DEFAULT_CONFIG_FILE
+    if 'PDIR2_CONFIG_FILE' in os.environ:
+        if os.path.exists(os.environ['PDIR2_CONFIG_FILE']):
+            os.remove(os.environ['PDIR2_CONFIG_FILE'])
+        del os.environ['PDIR2_CONFIG_FILE']
+    if os.path.exists(DEFAULT_CONFIG_FILE):
+        os.remove(DEFAULT_CONFIG_FILE)
+
+
```

Comment:
I think it will work. I made `import pdir` first then I found the way to clean cache, the first try is no required anymore i think. as the `fake_tty` will always clean the import caches.